### PR TITLE
[WIP] Skillchips

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds Skillchips, a highly requested feature. Skillchips are a organ that grants you a certain skill that allows you to do your job better, and inflicts brain damage when removed. 

The following skillchips will be implemented:
* Cablevision, available for Sci and Eng, allows you to see cables in doors/robots
* Nerveaugs, available for cargo and medbay, allows you to firemancarry faster, and pull crates faster
* Steady Hand skillchip, available for surgeons, chefs and engineers, and allows them to perform surgery and certain actions faster or more reliably (specifically, surgery, tools and butchery)
* Chem Skillchip, available to chemists and bartenders, allows them to use chem dispensers properly (chem dispensers will dispense larger, random amounts of reagents without this skillchip)
* Chef CQC Skillchip
* Security Martial Arts Skillchip

So this is my unique take on skillchips:
* Skillchips are multi purpose, and multiple jobs benefit from one skillchip (like the case of steady hands and nerve augs)
* Each department has min 2 skillchips that enhance their functionality
* Nobody starts with a skillchip, instead skillchips are dispensed early game and are available to print (CQC and sec chips are a bit more restricted)
* Removing a skillchip inflicts brain damage, but command would have a trait/implant that prevents this damage

Feedback and suggestions always welcome.

##Progress

- [x] Skillchip (basic organ) 
- [ ] Cablevision Skillchip
- [ ] Nerveaugs Skillchip 
- [ ] Surgery Skillchip
- [ ] Chem Skillchip
- [ ] Chef CQC Skillchip
- [ ] Martial Arts Skillchip
- [ ] Multi Skillchip Implant
- [ ] Skillchip Distribution (dispensers and printing)

## Changelog
:cl:
add: Multiple skillchips
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
oof not again

**Comment by ghost** [comment]
The chem skill chip is badly designed, it should be reworked in line with the other skill chips. None of the other chips break the related gameplay mechanic, in fact they only improve performance instead of being mandatory.

Change the skill chip to decrease chem dispenser charge cost instead, or have it unlock complex chem dispensing.

**Comment by Sinmero** [comment]
> Chem Skillchip, available to chemists and bartenders, allows them to use chem dispensers properly (chem dispensers will dispense larger, random amounts of reagents without this skillchip)

Bad

**Comment by Effemu** [comment]
sec martial art 😳 uh oh

**Comment by DatBoiTim** [comment]
#3242 Still Open???

**Comment by eeSPee** [comment]
> #3242 Still Open???

Oh shit I must have missed that one.

**Comment by utilbird** [comment]
What in the god damn 👀

File: code/modules/surgery/organs/augments_skillchips.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/obj/item/organ/cyberimp/skillchip/Insert(var/mob/living/carbon/M, var/special = 0, drop_if_replaced = FALSE)
+	. = ..()
+	if (trait)		
+		ADD_TRAIT(M, trait, SKILLCHIP_TRAIT)
+
+/obj/item/organ/cyberimp/skillchip/Remove(var/mob/living/carbon/M, special = FALSE)
+	if (!special)
+		M.setOrganLoss(ORGAN_SLOT_BRAIN, 25)
+	if (!trait)
```
Comment (Archanial) [inline code/modules/surgery/organs/augments_skillchips.dm:20]: shouldn't this be the other way around? If it has trait remove it?

**Comment by francinum** [comment]
Not this shit again

**Comment by PowerfulBacon** [comment]
🔑

**Pull request closed without merging**